### PR TITLE
Update Azure CLI reference summary pages

### PR DESCRIPTION
**Pull request by dksimpson**
This work is for story [1827767](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1827767). Remove deprecated references and add new ones, update core/extension status, and remove manual extension install instructions.

### CODE REVIEW

**Comment by dksimpson** [comment]
Hi @dbradish-microsoft, please review these updates I've made to the summary pages.

File: docs-ref-conceptual/azure-cli-reference-for-IoT.md [hunk 1]
```diff
@@ -37,57 +37,49 @@ The IoT **extension** introduces rich features and functionality to manage, mani
 
 | Reference | Has core | Description
```
Comment (dbradish-microsoft) [inline docs-ref-conceptual/azure-cli-reference-for-IoT.md:38]: The core and extension tables need to be combined as we no longer have separate lists.  The new TOC is not "extension type" aware and it doesn't make sense to provide two links to the same place.

**Review by dbradish-microsoft — CHANGES_REQUESTED** [review]
@dksimpson ,  once table format request for IoT and let's add a sentence or Note! before each table stating that "You will be prompted to install an extension reference on first use.  You can also use the [az extension add](/cli/azure/extension#az_extension_add) command to manually install an extension."  Please format and wordsmith as you think best.

File: docs-ref-conceptual/azure-cli-reference-for-IoT.md (hunk 1 above)
Comment (dksimpson) [reply docs-ref-conceptual/azure-cli-reference-for-IoT.md:38]: Hi @dbradish-microsoft, what does "once table format request for IoT" mean?

**Comment by dksimpson** [comment]
Hi @dbradish-microsoft, I've incorporated your review comments.

**Pull request merged**